### DISCUSSION TITLE
Adding different colors in MACD lines

### DISCUSSION THREAD

**Discussion by andreltr** [serkor1/cryptoQuotes#30] [General]
Hello,

Would it be possible to customize nFast and nSlow lines colors to a given pair, for instance red for nFast line and purple for nSlow.  Would it be similar to a line such as cryptoQuotes::sma(n = 33, price = "close", color = "blue")?  How about their width?

cryptoQuotes::macd(nFast = 12,
                       nSlow = 21,
                       nSig = 9,
                       maType = "SMA",
                       percent = TRUE)

Thanks!

**Comment by distortedMango** [comment]
Have you tried the development version @andreltr? There is a greater flexibility on the color theme of `chart()`!

**Reply by andreltr** [reply]
No, I haven't. My R programming expertise is limited somewhat I confess.

**Reply by distortedMango** [reply]
Well, give it a go! 😄 

You can install the development version as follows,

``` r
## install from github
devtools::install_github(
  repo = 'https://github.com/serkor1/cryptoQuotes/',
  ref  = 'development'
)
```

If you are looking for good `R` ressources I would strongly suggest the [Deep R Programming](https://github.com/gagolews/deepr)-book. Give the guy a star and a fork to show appreciation! 😄

**Reply by serkor1** [reply]
Considering your skill in `R`, I will not recommend a hacky solution using {plotly}. I will instead recommend the development version too, which has a new feature where the colors can be customized on the entire chart.

**Reply by serkor1** [reply]
@andreltr if the development version is not enough, or not what you are looking for - then provide me a usecase for ability of coloring MACD.

It already has a color deficiency usecase, so I am a bit out of ideas on what to do with the MACD and the coloring, if you catch my meaning.

**Reply by andreltr** [reply]
> @andreltr if the development version is not enough, or not what you are looking for - then provide me a usecase for ability of coloring MACD.
> 
> It already has a color deficiency usecase, so I am a bit out of ideas on what to do with the MACD and the coloring, if you catch my meaning.

Hello,

Thanks for the message.  My idea is for the user being able to customize the colors of lines ema, wma, Bollinger bands or MACD, for instance,

    cryptoQuotes::ema(n = 13, price = "close", color = "darkorchid1"),  
    cryptoQuotes::sma(n = 33, price = "close", color = "blue"),    
    cryptoQuotes::wma(n = 55, price = "close", color = "green")    

in an example for the macd,

    cryptoQuotes::macd(nFast = 12, color = "",
                       nSlow = 21, color = "",
                       nSig = 9, color = "",
                       maType = "SMA",
                       percent = TRUE)

One thing that could also help would be to easily change the lines thickness for it could help people with reading problems, IMHO.

Best regards,

Andre Luiz

**Reply by serkor1** [reply]
@andreltr this is actually a really good point that I have not considered. 

I will add a functionality similar to that of 'scale' which I have already implemented for all lines.

Thank you so much for the feedback. I will get to it ASAP! 🫡

**Reply by serkor1** [reply]
Hi @andreltr,

Firstly, thank you for pointing out the issue with reading the line thickness in relation to people with reading problems. I have added a new option in the `chart()`-function called `width` which does what you asked for. Below is MWE,

## MWE

``` r
library(cryptoQuotes)
```

``` r
# default chart
chart(
  ticker = BTC,
  indicator = list(
    sma()
  ),
  sub    = list(
    macd()
  )
)
```

![](https://i.imgur.com/ReADs3A.png)

``` r
# with increased width
chart(
  ticker = BTC,
  indicator = list(
    sma()
  ),
  sub    = list(
    macd()
  ),
  options = list(
    width = 2
  ) 
)
```

![](https://i.imgur.com/PbZufFW.png)

<sup>Created on 2024-08-16 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Reply by andreltr** [reply]
Hello,

The solution is OK, IMHO.
I wonder if you are considering the idea of allowing customization of the axes colors in the macd() function parameters
in nFast, nSlow, nSig and ema(), sma(), and wma() functions.  Thanks.

Great job!

**Reply by serkor1** [reply]
> Hello,
> 
> 
> 
> The solution is OK, IMHO.
> 
> I wonder if you are considering the idea of allowing customization of the axes colors in the macd() function parameters
> 
> in nFast, nSlow, nSig and ema(), sma(), and wma() functions.  Thanks.
> 
> 
> 
> Great job!

Thank you! Well, I have asked you for a usecase like the one about the linewidths - and you didn't give me one yet. So no, currently its not something that I am considering.

The idea behind the chart()-function is that it should be high-level. Might be that its a bad idea, of so please convince me!

I am open to well argued ideas 😁
